wit/atomic.wit

```diff
@@ -19,4 +19,12 @@ interface atomics {
 	///
 	/// If any other error occurs, it returns an `Err(error)`.
 	increment: func(bucket: borrow<bucket>, key: string, delta: u64) -> result<u64, error>;
+
+	/// Atomically set the value associated with the key in the store to the given value. It returns
+	/// `true` if the value was set properly, or `false` if the swap failed.
+	/// 
+	/// If setting a new value, a `none` value should be passed for the `old` parameter. If the key
+	/// already exists in the store, it will be swapped with the new value only if the current value
+	/// is equal to the old value. To delete a key-value pair, pass `none` for the `new` parameter.
+	compare-and-swap: func(key: string, old: option<list<u8>>, new: option<list<u8>>) -> result<bool, error>;
```

Comment:
From the description, it seems like cas is pretty complicated. Would it make more sense to break it down to a few functions that each does one thing?

1) `create: func(key: string, value: list<u8>) -> result<bool, error>` will succeed only if there is no current value associated this key. (in the case where `old` is `none` in compare-and-swap)
2) `delete: func(key: string, value: list<u8>) -> result<bool, error>` will succed only if the value associated with the key is the `value`.
3) `compare_and_swap`: func(key: string, old: list<u8>, new: list<u8>) -> result<bool, error>;

Reply:
I would prefer the single function to keep the breaking changes/additions easier to manage and I also think this matches the semantics of at least a few of the CAS things I've seen before. Curious if anyone else has opinions/experience here as well who could weigh in

Reply:
Agreed, let's keep it one function.